ci-operator/step-registry/openshift/upgrade/openstack/loki/openshift-upgrade-openstack-loki-workflow.yaml

```diff
@@ -17,9 +17,6 @@ workflow:
       TEST_TYPE: upgrade
       TEST_SUITE: ""
       TEST_UPGRADE_OPTIONS: ""
-      OPENSTACK_EXTERNAL_NETWORK: "external"
-      OS_CLOUD: "openstack-cloud"
-      OPENSTACK_COMPUTE_FLAVOR: "m1.s2.xlarge"
```

Comment:
all this changes might seem valuable, but they are just moving lines of code. 
They were added one at at time when the workflow file was created. 
And what's more, by leaving them here I can emediately tell what this workflow (test) is being run on. 
openstack-cloud (MOC), with m1.s2.xlarge flavor and on the external network. 

If you remove them from here, and later on I decide I want to run it on a m1.s3.xlarge flavor, how would I do that without affecting other tests (workflows)?

Reply:
> If you remove them from here, and later on I decide I want to run it on a m1.s3.xlarge flavor, how would I do that without affecting other tests (workflows)

`ipi-conf-openstack` can be modified so that it knows more flavor names, and selects them based on new `env` variables that define the desired RAM quantity or disk space.

In your scenario:

1. Let's say we move from `m1.s2.xlarge` to `m1.s2.medium` to test with less RAM.
2. Then we move to Vexxhost. We have to change the same variable to an arbitrary different flavor name, just to keep the test as-is
3. Then we need more disk space: we have to change again the same variable

In all this process, you can only know what we are doing if you have access to MOC. Otherwise, there is now way to know that `m1.s2.medium` has exactly 4GB RAM. Essentially, the knowledge about the flavors stays in your mind (or in your computer). In my scenario, the knowledge about the flavors is embedded in one single mapping file (`ipi-conf-openstack`).

Reply:
having to modifye ip-conf-openstack just so I can define a new workflow does not sound optimal. 
Maybe ipi-conf-openstack can be made to accept more generic inputs from ref. editing the *.sh file of the step is not good for maintanance.

Reply:
> having to modifye ip-conf-openstack just so I can define a new workflow does not sound optimal

You don't have to touch `ipi-conf-openstack` to define a new workflow.